### ISSUE TITLE
bug: update docs and logging on chain support for ENS

### ISSUE THREAD

**Issue by bitbeckers** [wagmi-dev/wagmi#2097]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.12.7

### Current Behavior

When using `useEnsName` the hook returns `null` or `undefined` when the application is not on an ENS supported name (in my case Goerli).

The issue is:
- It's not documented that that needs to be, because I can also pass `chainId: 1` in the parameters according to your docs
- There are no errors or warnings returned in the `data`
- The hook does not resolve/callback using the `onSuccess`, `onSubmitted`, `onError` variables in the callback

When I add `mainnet` to the supported network list of `configureChains` it immediately resolves without setting `chainId: 1` in the `useEnsName` hook. But only for the ConnectButton and not for components using the hook directly; that required `chainId: 1`.

TLDR: Using `useEnsName` is a little black-box

### Expected Behavior

- Clear documentation on requiring an ENS supported network in the chainConfig for ENS to resolve
- Warnings or errors when ENS cannot resolve due to unsupported chains

### Steps To Reproduce

```js
const { chains, provider } = configureChains(
  [goerli], //or [goerli, mainnet]
  [alchemyProvider({ apiKey: ALCHEMY_API_KEY! }), publicProvider()]
);
```

```js
  const {
    data: _avatar,
    isError: avatarError,
    isLoading: avatarLoading,
  } = useEnsAvatar({
    address,
   //chainId: 1
  });
  ```

### Anything else?

Happy to work on this and make a PR, starting with the documentation, but I wanted to reach out first.

**Comment by tmm** [comment]
ENS only works if you are using a [support chain](https://docs.ens.domains/ens-deployments). Agree that we could bubble up an error if ENS is not supported on the chain. This is actually already implemented in [viem](https://viem.sh) and will start working once we migrate away from ethers (see #2135).
